### ISSUE TITLE
Consensus fail for payment transaction

### ISSUE THREAD

**Issue by kfangw** [bosnet/sebak#150]
I am trying send coin via wallet.
bellow is my step

prepare docker image and containers
```
docker build . -t sebak && docker rmi -f $(docker images -f "dangling=true" -q)

docker run --net=host --rm -it --env-file=docker/node1.env sebak
docker run --net=host --rm -it --env-file=docker/node2.env sebak
docker run --net=host --rm -it --env-file=docker/node3.env sebak
```

generate 3 key pairs
```
[~/goworkspace/src/boscoin.io/sebak] sebak key generate
       Secret Seed: SBRJ7NFNW6Q7CKGLILMEWHAZX353EKV2HXOWMHS7UGTBOQGH2OM5J6DS
    Public Address: GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y
[~/goworkspace/src/boscoin.io/sebak] sebak key generate
       Secret Seed: SDKZB4CROCE4OJIFS4ZX6BIBD7TKS7BEGTFLYBKS2CV24JSJA3WJKBSM
    Public Address: GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4
[~/goworkspace/src/boscoin.io/sebak] sebak key generate
       Secret Seed: SBO3SK4AXR7ANLAOKNAVJLE23I2MT6ZTJTOWZGS7AJ6WFMCL764HVPXC
    Public Address: GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED
```

send 10000000 to 3 account
```
docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create ;docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create ;docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create
```

prepare another docker container to curl
```
docker run --net host --rm -it ubuntu:14.04 bash
apt-get update
apt-get install curl
```

in the docker container call the account api
```
echo "====1====" ;curl -k https://127.0.0.1:2823/api/account/GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y ;echo ;echo "====2====";curl -k https://127.0.0.1:2823/api/account/GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4 ;echo ;echo "====3====" ;curl -k https://127.0.0.1:2823/api/account/GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED ; echo
```
==>
```
====1====
{"Address":"GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y","Balance":"10000000","Checkpoint":"5nc4PvPKMxDTM6ets3hudktkA-5nc4PvPKMxDTM6ets3hudktkA"}
====2====
{"Address":"GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4","Balance":"10000000","Checkpoint":"9oJ7Y2xU6oCPhEQkrC9uikeLai38B5uFU3ZLyBc5hhLm-9oJ7Y2xU6oCPhEQkrC9uikeLai38B5uFU3ZLyBc5hhLm"}
====3====
{"Address":"GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED","Balance":"10000000","Checkpoint":"Aq6yzC8VraFAQnQ3DuBEFwDHZQCoZmnQtRKokJ9LMcjU-Aq6yzC8VraFAQnQ3DuBEFwDHZQCoZmnQtRKokJ9LMcjU"}
```

each account have balance 1000

now I tried send from 1->3
```
docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED 130000 SBRJ7NFNW6Q7CKGLILMEWHAZX353EKV2HXOWMHS7UGTBOQGH2OM5J6DS --endpoint=https://127.0.0.1:2822
```
2->3
```
docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED 120000 SDKZB4CROCE4OJIFS4ZX6BIBD7TKS7BEGTFLYBKS2CV24JSJA3WJKBSM --endpoint=https://127.0.0.1:2822
```

3->2
```
docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4 120000 SBO3SK4AXR7ANLAOKNAVJLE23I2MT6ZTJTOWZGS7AJ6WFMCL764HVPXC --endpoint=https://127.0.0.1:2822
```

Check the balance again
```
docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create ;docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create ;docker build . -t sebak && docker run --rm --network host -it sebak wallet payment GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED 10000000 SBECGI3FSCYHNQIMANNCWQSVA6S5C6L4BXFKAPMBAMI5V47NWXNE37MN --endpoint=https://127.0.0.1:2822 --create
```
==>
```
====1====
{"Address":"GCXQHQDSU6DHNW3YV2QXGEFERQ3E4WIHMVUIWBRGAGT7765WOPUHW54Y","Balance":"9860000","Checkpoint":"F3RoU6TkGe1Wrn8bd4fzZNtoBkZAQqbF8gpk6218CmKW-F3RoU6TkGe1Wrn8bd4fzZNtoBkZAQqbF8gpk6218CmKW"}
====2====
{"Address":"GB6UZVTLJF545XA4ZXDGH2AI7RD62YVIRL7H5MCPHTXZ4BOHOPUSLCM4","Balance":"10000000","Checkpoint":"9oJ7Y2xU6oCPhEQkrC9uikeLai38B5uFU3ZLyBc5hhLm-9oJ7Y2xU6oCPhEQkrC9uikeLai38B5uFU3ZLyBc5hhLm"}
====3====
{"Address":"GA2GII5TSFTVDHS23HBVYEHOLFNGATUPRNDJMVD2FWA4ZOI3B2TSAHED","Balance":"10130000","Checkpoint":"5nc4PvPKMxDTM6ets3hudktkA-F3RoU6TkGe1Wrn8bd4fzZNtoBkZAQqbF8gpk6218CmKW"}
```

`it seems like that Only first account can get/send coins to another account`

Bellows is log message when 2->3 failed
```
DBUG[06-29|06:08:49] got message from client`                 module=sebak node=GAYG.TLY7 message="{\"Type\":\"message\",\"Data\":\"eyJUIjoidHJhbnNhY3Rpb24i"
DBUG[06-29|06:08:49] message is transaction                   module=sebak node=GAYG.TLY7
DBUG[06-29|06:08:49] saved in history                         module=sebak node=GAYG.TLY7 transaction=G583PEDTJQc5Ez4rrYkfWmNk3WLwvQp24P23J8iRaP8h
DBUG[06-29|06:08:49] ballot from client will be broadcasted   module=sebak node=GAYG.TLY7 ballot=G583PEDTJQc5Ez4rrYkfWmNk3WLwvQp24P23J8iRaP8h
127.0.0.1 - - [29/Jun/2018:06:08:49 +0000] "POST /node/ballot HTTP/2.0" 200 1006 "" "v-GDTE.6RAE"
DBUG[06-29|06:08:49] got ballot                               module=sebak node=GAYG.TLY7 message="{\"Type\":\"ballot\",\"Data\":\"eyJUIjoiYmFsbG90IiwiSCI6e"
127.0.0.1 - - [29/Jun/2018:06:08:49 +0000] "POST /node/ballot HTTP/2.0" 200 1006 "" "v-GDIR.M6CC"
DBUG[06-29|06:08:49] got ballot                               module=sebak node=GAYG.TLY7 message="{\"Type\":\"ballot\",\"Data\":\"eyJUIjoiYmFsbG90IiwiSCI6e"
DBUG[06-29|06:08:49] check threshold                          module=sebak state=INIT threshold=3 yes=3 no=0 policy="{\"accept\":60,\"init\":100,\"sign\":60,\"validators\":3}"
DBUG[06-29|06:08:49] ballot will be broadcasted               module=sebak node=GAYG.TLY7 ballot=G583PEDTJQc5Ez4rrYkfWmNk3WLwvQp24P23J8iRaP8h isNew=false
127.0.0.1 - - [29/Jun/2018:06:08:49 +0000] "POST /node/ballot HTTP/2.0" 200 1005 "" "v-GDTE.6RAE"
DBUG[06-29|06:08:49] got ballot                               module=sebak node=GAYG.TLY7 message="{\"Type\":\"ballot\",\"Data\":\"eyJUIjoiYmFsbG90IiwiSCI6e"
DBUG[06-29|06:08:49] check threshold                          module=sebak state=SIGN threshold=2 yes=0 no=2 policy="{\"accept\":60,\"init\":100,\"sign\":60,\"validators\":3}"
DBUG[06-29|06:08:49] consensus of this ballot will be closed  module=sebak ballot=G583PEDTJQc5Ez4rrYkfWmNk3WLwvQp24P23J8iRaP8h
DBUG[06-29|06:08:49] consensus closed                         module=sebak node=GAYG.TLY7
127.0.0.1 - - [29/Jun/2018:06:08:49 +0000] "POST /node/ballot HTTP/2.0" 200 1005 "" "v-GDIR.M6CC"
DBUG[06-29|06:08:49] got ballot                               module=sebak node=GAYG.TLY7 message="{\"Type\":\"ballot\",\"Data\":\"eyJUIjoiYmFsbG90IiwiSCI6e"
WARN[06-29|06:08:49] failed to add VotingResult               module=sebak MessageHash=G583PEDTJQc5Ez4rrYkfWmNk3WLwvQp24P23J8iRaP8h error="{\"code\":121,\"message\":\"ballot has invalid state\"}"
EROR[06-29|06:08:49] failed to handle ballot                  module=sebak node=GAYG.TLY7 error="{\"code\":114,\"message\":\"`VotingResult` not found\"}"
127.0.0.1 - - [29/Jun/2018:06:08:50 +0000] "POST /node/connect HTTP/2.0" 200 265 "" "v-GDTE.6RAE"
```

**Comment by anarcher** [comment]
During my test,  nodes seem to like to stop when their get messages. 
Is it only my condition? 

```

DBUG[06-29|06:11:15] got message from client`                 module=sebak node=GDTE.6RAE message="{\"Type\":\"message\",\"Data\":\"eyJUIjoidHJhbnNhY3Rpb24i"
DBUG[06-29|06:11:15] message is transaction                   module=sebak node=GDTE.6RAE


DBUG[06-29|06:11:34] got message from client`                 module=sebak node=GDTE.6RAE message="{\"Type\":\"message\",\"Data\":\"eyJUIjoidHJhbnNhY3Rpb24i"
DBUG[06-29|06:11:34] message is transaction                   module=sebak node=GDTE.6RAE


DBUG[06-29|06:12:29] got message from client`                 module=sebak node=GDTE.6RAE message="{\"Type\":\"message\",\"Data\":\"eyJUIjoidHJhbnNhY3Rpb24i"
DBUG[06-29|06:12:29] message is transaction                   module=sebak node=GDTE.6RAE
```

**Comment by kfangw** [comment]
Solved by #152
